docs/docs/general/03-Element Selection/recommended-practices.md

```diff
@@ -0,0 +1,171 @@
+---
+sidebar_position: 2
+title: Recommended Practices
+---
+
+# Recommended Practices
+On this page you will learn the different methods to detect elements. Every method has its strength and weaknesses, for example one is more accurate than another but also more effort when using it.
+
+## The Overarching Goal
+As a user you want to detect elements reliably. There are two axes of reliability:
+
+* False Positives: Detecting the wrong element or detecting more elements than you want to detect.
+* False Negatives: Detecting no element where there is one.
+
+## Methods of Detecting Elements in AskUI
+
+**Ranked by detection accuracy:**
+
+| Method         | Description                               |   Advantages     | Disadvantages                            |
+| :------------- | :-------------                            | :--------------- | :---------------                         | 
+| custom element | Detects elements by image-in-image search | Reliable         | Elaborate creation |
+| text           | Finds an element of class text. If you pass an argument `text('hello')` it also uses text-recognition | Fast to write. High reliability | Dependent on resolution, font and linebreaks |
+| button, textfield, checkbox, ...| Finds an element of the class      | Reliable when element is clearly distinguishible | Sometimes fails if element, for example a button, is clearly distinguishable as a specific class |
+| icon           | Finds elements of type icon and then filters for _specific_ icon | Detects icons reliably | Naming of icons inconsistent. |
+| matching       | Natural language description.             | Works on logos and images with text; Needs experimentation for detecting the correct element | Only finds the best matching element |
+
+## Recommendation for Optimal Detection Experience
+The general recommendation is to use a class-based method like `text`, `button` or `textfield` first. When you run into non-detectable elements, like a button AskUI does not detect, you can use a custom element.
+
+Here are two problems to consider while implementing workflows:
+
+* Responsive text causes linebreaks: Custom element detection and text detection fails.
+* The color of element/text changes during workflow.
```

Comment:
Let's integrate it in a way where we first present the different detection options (removing disadvantages and advantages as there are no or not a lot general/absolute advantages/disadvantages) and after that do a use-case specific presentation with the two points here being 2 of the use cases.